### PR TITLE
Updates to work with the new FleetSensor API

### PR DESCRIPTION
**Pull request by fjarri**
In https://github.com/nucypher/nucypher/pull/2352 `FleetSensor` and node metadata API changed. This PR updates the calls in the monitor. Needs https://github.com/nucypher/nucypher/pull/2574 to work, so the CI will likely fail...

There is also a change in how crawler storage is implemented. Before it was derived from `ForgetfulNodeStorage`, which presented several problems:
- crawler stores node statuses and fleet states, while `ForgetfulNodeStorage` stores node metadata, so the hierarchical relation between them is questionable
- node status cannot be obtained from a node object itself, so the crawler storage simply did not have access to enough data

In this PR, `CrawlerStorage` is its own class, and its calls are hooked to `FleetSensor` updates (see `hooked_tracker_class()`). 

**Important:** One consequence of this is that we now *do not* store nodes right away before they are even verified; now the nodes are stored (or removed) when the fleet state changes. This can be modified, but currently I am unclear as to what behavior was really expected from the crawler, so I went with the logical one.

### CODE REVIEW

File: monitor/crawler.py
Comment (fjarri) [reply monitor/crawler.py:166]: I'm starting to think it might be better to return an actual documentable object from `status_info()`, with a `to_json()` method. Using JSON for interfacing inside Python seems like a bad idea.

> Looks like characters length is 2

Nope, state nickname is only 1 character. I have to do this since I don't want to change the DB schema, but in general it would be better to make it length-agnostic.

Comment (KPrasch) [reply monitor/crawler.py:166]: Sounds like a job for a named tuple.

Comment (fjarri) [reply monitor/crawler.py:166]: We do need it to be json-serializable, but we can just overload the named tuple and add a method. Perhaps a dataclass would fit better, but we have to support Py3.6

Comment (KPrasch) [reply monitor/crawler.py:166]: Yes, you're right. I don't forsee any issues supporting 3.6+ only.

Comment (fjarri) [reply monitor/crawler.py:166]: I meant, dataclasses require 3.7+ and we support 3.6+. But making a derived class from a namedtuple shouldn't be a problem.

Comment (KPrasch) [reply monitor/crawler.py:166]: Ah, right. Well same goes for 3.7+.  In fact CI is already using that version.  No strong preference though, it's the namespacing that matters most I think.

Comment (derekpierre) [reply monitor/crawler.py:166]: > Nope, state nickname is only 1 character. I have to do this since I don't want to change the DB schema, but in general it would be better to make it length-agnostic.

Whoops read the wrong Nickname usage. Yes, my hope was to figure out a way to make it length agnostic

File: monitor/crawler.py
Comment (fjarri) [reply monitor/crawler.py]: Yep, fixed.

File: monitor/crawler.py
Comment (fjarri) [reply monitor/crawler.py:110]: Actually I now realize that the DB here is transient, so we can change the schema. Maybe we should.

**Review by derekpierre — COMMENTED** [review]
Looks good. Before merging, we'll need to relock dependencies once there is a `nucypher` release with https://github.com/nucypher/nucypher/pull/2574 in it.

Presumably, you've run it locally with a `nucypher` version containing #2574?

**Comment by fjarri** [comment]
> Presumably, you've run it locally with a nucypher version containing #2574?

I did, but to be on the safe side someone else should too.

**Comment by derekpierre** [comment]
Ran it locally with relocked dependencies (pointed to `nucypher` on `main`), and it seemed to run fine. The scraping round took some time (9-10mins), but I usually find that it takes longer on my local machine, and ends up being much faster on the server (~2mins).

**Comment by fjarri** [comment]
Current status: waiting for the new release with merged #2574 so that CI could run

File: monitor/crawler.py
Comment (fjarri) [reply monitor/crawler.py]: Fixed.

**Comment by fjarri** [comment]
With `nucypher` updated to 4.8, tests pass now.

**Review by KPrasch — APPROVED** [review]
Thanks for maintaining the monitor! 😄

**Comment by fjarri** [comment]
Actually the dependency relock can probably be rolled back to minimize the changes. nucypher 4.8 will be installed anyway.

**Review by derekpierre — APPROVED** [review]
🎸

**Pull request merged**
